Pid cleaning

alrik-durand opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
I've changed the process_interface and process_control_interface to respect Qudi's naming conventions.
I've updates the hardware module accordingly.
I've added the unit getters fot the pid_controller_interface.

## Motivation and Context
These interfaces are not used a lot yet. I've updated them before future development.

## How Has This Been Tested?
I've tested this on a new code using software_pid_controller (to come). I have not tested the modified hardware modules as I don't have the hardware to test it.

## Types of changes
<!--- What types of changes does your code introduce? Put an 'x' in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
<!--- Go over all the following points, and put an 'x' in all the boxes that apply. -->
<!--- If you're unsure about any of these, ask. -->
- [x] My code follows the code style of this project.
- [ ] I have documented my changes in the changelog (`documentation/changelog.md`)
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have added/updated for the module the config example in the docstring of the class accordingly.
- [x] I have checked that the change does not contain obvious errors (syntax, indentation, mutable default values).
- [ ] I have tested my changes using 'Load all modules' on the default dummy configuration with my changes included.
- [ ] All changed Jupyter notebooks have been stripped of their output cells.

kay-jahnke commented:
Hi @alrik-durand : 
We are in the process of cleaning up the qudi PRs. As the cleaning of the interface should happen together with the implementation of the hardware, we would like to close this PR for now.
Please resubmit the code together with your hardware implementation.

signed: @Neverhorst @drogenlied @kay-jahnke

alrik-durand commented:
I've added some of the code that goes with these changes.

I don't think changing the interfaces is absolutely required. As there was very few module using them, I thought it would be a good idea to do it now instead of later/never.

The code I've added probably deserve it's own PR but it shows where I was going with this !
